mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -619,6 +618,88 @@ func resourceContainerCluster() *schema.Resource {
 											},
 										},
 									},
+									"upgrade_settings": {
+										Type:             schema.TypeList,
+										Optional:         true,
+										Description:      `Specifies the upgrade settings for NAP created node pools`,
+										Computed:         true,
+										MaxItems:         1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"max_surge": {
+													Type:        schema.TypeInt,
+													Optional:    true,
+													Description: `The maximum number of nodes that can be created beyond the current size of the node pool during the upgrade process.`,
+												},
+												"max_unavailable": {
+													Type:        schema.TypeInt,
+													Optional:    true,
+													Description: `The maximum number of nodes that can be simultaneously unavailable during the upgrade process.`,
+												},
```

Comment:
Looks like we remove the `computed` from these two field. I wonder if the API will return values for these two fields if we don't explicitly specify them for surge upgrade?

Reply:
The max_surge and max_unavailable values in this case have been verified to be only necessary for the SURGE. The strategy from SURGE to BLUE GREEN cannot be updated if we leave this attrs in its computed state since d.GetOk will utilise its last stetted value if its values are not present in the current tf file. As a result, even though we didn't include it in the tf file, it will assume that we have provided its (max_surge/max_unavailable) value and initiate the validation.

The ideal scenario would be to enable us to switch from an SURGE strategy to a BLUE GREEN strategy by removing the SURGE attrs (max surge and max unavailable) and adding the BLUE GREEN settings.

In a different instance, API returned zero values and set them into a state file for the blue_green_settings(which not having computed:True) and resulted in an indication that changes needed to be updated in the subsequent apply even though there were no changes in the TF file. I have added computed:True for it in the next commit.

max_surge and max_unavailable, however, are of the Int type. Due to this, it will consider computed to be zero and set its value to 0 if we don't supply computed as True.